src/battle-dex-search.ts

```diff
@@ -1256,36 +1262,55 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
 			return moves.includes('stickyweb');
 		case 'irontail':
 			return dex.gen > 5 && !moves.includes('ironhead') && !moves.includes('gunkshot') && !moves.includes('poisonjab');
+		case 'jawlock':
+			return species.baseSpecies === 'Drednaw';
 		case 'jumpkick':
 			return !moves.includes('highjumpkick');
 		case 'leechlife':
 			return dex.gen > 6;
+		case 'metalburst':
+			return abilityid === 'sturdy';
 		case 'mysticalfire':
 			return dex.gen > 6 && !moves.includes('flamethrower');
 		case 'naturepower':
 			return dex.gen === 5;
+		case 'nightdaze':
+			return dex.gen === 5 && species.baseSpecies === 'Zoroark';
 		case 'nightslash':
 			return !moves.includes('crunch') && !(moves.includes('knockoff') && dex.gen >= 6);
 		case 'petaldance':
 			return abilityid === 'owntempo';
 		case 'phantomforce':
 			return (!moves.includes('poltergeist') && !moves.includes('shadowclaw')) || this.formatType === 'doubles';
+		case 'poisonfang':
+			return dex.gen > 5 && species.types.includes('Poison');
+		case 'pollenpuff':
+			return !moves.includes('bugbuzz')
+		case 'powertrip':
+			return species.baseSpecies === 'Incineroar';
 		case 'relicsong':
 			return species.id === 'meloetta';
 		case 'refresh':
 			return !moves.includes('aromatherapy') && !moves.includes('healbell');
+		case 'reversal':
+			// SubSalac / EndSalac Heracross in Gen 3
+			return dex.gen === 3 && species.baseSpecies === 'Heracross';
```

Comment:
In case it wasn't clear by now, I prefer not to hardcode species. Maybe this should be any Fighting type?